### PR TITLE
Enable dependabot to automatically get dependency updates

### PR DESCRIPTION
**Pull request by MRamonLeon**
It would be great to have **Dependabot** configured to get dependencies updates filed automatically. This will help avoiding bugs or security vulnerabilities from old dependencies.

Some helpful documentation about Dependabot to fine adjust it if needed:

* Keeping your dependencies updated automatically: https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/keeping-your-dependencies-updated-automatically
* Avoiding security vulnerabilities: https://docs.github.com/en/free-pro-team@latest/github/managing-security-vulnerabilities/configuring-github-dependabot-security-updates
* Fine adjusting the configuration: https://docs.github.com/en/free-pro-team@latest/github/administering-a-repository/configuration-options-for-dependency-updates

### CODE REVIEW

**Comment by batmat** [comment]
retriggering...

**Review by oleg-nenashev — CHANGES_REQUESTED** [review]
Please add filtering for Jenkins core dependencies here. Requesting the change as a former maintainer. When the plugin is adopted, a new maintainer is welcome to ignore the comment

**Comment by MRamonLeon** [comment]
@oleg-nenashev is it fine to you to be merged now? Thanks

**Review by oleg-nenashev — APPROVED** [review]
Fine with me, but I will leave it unmerged until a new maintainer steps in. I am not ready to process automatic dependency update requests for this component.

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 2
+updates:
+  - package-ecosystem: "maven"
+    directory: "/"
+    schedule:
+      interval: "weekly"
+    ignore:
+    - dependency-name: org.jenkins-ci.main:jenkins-core
+      versions:
+      - ">= 0"
```
Comment (jglick) [inline .github/dependabot.yml:10]: Not actually necessary—it is not offered anyway.

**Review by jglick — COMMENTED** [review]
Should be feasible in general after #28.

**Comment by jglick** [comment]
Build failure due to EC2 Windows agent problem; perhaps better to `useAci: true`.

**Review by jglick — APPROVED** [review]
I have GitHub and Artifactory permissions, though I would not want to be making any nontrivial changes here since I do not have a test environment to sanity-check anything.

**Comment by MarkEWaite** [comment]
Closing this pull request because the plugin will be deprecated in May 2023.

**Pull request closed without merging**